### PR TITLE
Add apparent-horizon boundary conditions to XCTS executable

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Also includes some preparatory commits to slice the non-linear fields and fluxes to external boundaries.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Elliptic/DiscontinuousGalerkin/Initialization.cpp [hunk 1]
```diff
@@ -97,6 +107,38 @@ void InitializeFacesAndMortars<Dim>::operator()(
...
+  // Compute the Jacobian derivative numerically, because our coordinate maps
+  // currently don't provide it analytically.
+  if (not element.external_boundaries().empty()) {
+    ASSERT(mesh.quadrature(0) == Spectral::Quadrature::GaussLobatto,
+           "Slicing the Hessian to the boundary currently supports only "
+           "Gauss-Lobatto grids. Add support to "
+           "'elliptic::dg::InitializeFacesAndMortars'.");
+    using inv_jac_tag =
+        domain::Tags::InverseJacobian<Dim, Frame::Logical, Frame::Inertial>;
+    Variables<tmpl::list<inv_jac_tag>> vars_to_derive{
```
Comment (kidder) [inline src/Elliptic/DiscontinuousGalerkin/Initialization.cpp]: vars_to_derive -> vars_to_differentiate

File: src/Elliptic/DiscontinuousGalerkin/Initialization.hpp [hunk 2]
```diff
@@ -70,31 +71,46 @@ struct InitializeGeometry {
...
-          Dim, tmpl::list<domain::Tags::Direction<Dim>,
-                          domain::Tags::Coordinates<Dim, Frame::Inertial>,
-                          domain::Tags::FaceNormal<Dim>,
-                          domain::Tags::UnnormalizedFaceNormalMagnitude<Dim>>>,
+          Dim,
+          tmpl::list<domain::Tags::Direction<Dim>,
+                     domain::Tags::Coordinates<Dim, Frame::Inertial>,
+                     domain::Tags::FaceNormal<Dim>,
+                     domain::Tags::UnnormalizedFaceNormalMagnitude<Dim>,
+                     // Possible optimization: The Hessian could be omitted for
```
Comment (kidder) [inline src/Elliptic/DiscontinuousGalerkin/Initialization.hpp]: Hessian -> derivative of the normal

**Review by kidder — COMMENTED** [review]
you can squash

**Review by kidder — DISMISSED** [review]

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
@wthrowe @nilsdeppe do either of you want to look at this?

**Comment by nilsdeppe** [comment]
I don't need to look at this. I glanced at the parts that aren't specific to the elliptic code, seems fine to me :)

**Review by wthrowe — APPROVED** [review]

**Pull request merged**